app/src/main/java/com/example/tanya/godeltechchallengeandroid/data/DataModule.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.tanya.godeltechchallengeandroid.data
+
+import android.content.res.Resources
+import com.example.tanya.godeltechchallengeandroid.R
+import com.example.tanya.godeltechchallengeandroid.data.prefs.ApplicationRepositoryImpl
+import com.example.tanya.godeltechchallengeandroid.util.Timespan
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+
+@Module
+abstract class DataModule {
```

Comment:
Also you can simplify the module. Please see my comment for the `ApiModule` class